homework6/Game/Game/Game.cs

```diff
@@ -0,0 +1,168 @@
+                        throw new InvalidDataException("There are more than one main character on the map.");
+                    }
+                }
+            }
+        }
+
+        if (!isMainCharacterDetectedOnMap)
+        {
+            throw new InvalidDataException("Main character is not found in the map.");
+        }
+        
+        return mainCharacterStartCoordinates;
+    }
+
+    private enum Direction
+    {
+        Left,
+        Right,
+        Up,
+        Down
+    }
+
+    private bool CanMoveInThisDirection(Direction direction)
+    {
+        switch (direction)
+        {
+            case Direction.Left:
+                return MainCharacterCoordinates.y > 0 &&
+                       map[MainCharacterCoordinates.x, MainCharacterCoordinates.y - 1] == freeSpace;
+            case Direction.Right:
+                return MainCharacterCoordinates.y < map.GetLength(1) - 1 &&
+                       map[MainCharacterCoordinates.x, MainCharacterCoordinates.y + 1] == freeSpace;
+            case Direction.Up:
+                return MainCharacterCoordinates.x > 0 &&
+                       map[MainCharacterCoordinates.x - 1, MainCharacterCoordinates.y] == freeSpace;
+            default: // Direction.Down
+                return MainCharacterCoordinates.x < map.GetLength(0) - 1 &&
+                       map[MainCharacterCoordinates.x + 1, MainCharacterCoordinates.y] == freeSpace;
+        }
+    }
+
+    private void MoveMainCharacter(Direction direction)
+    {
+        if (CanMoveInThisDirection(direction))
+        {
+            Console.SetCursorPosition(MainCharacterCoordinates.y, MainCharacterCoordinates.x);
+            Console.WriteLine(freeSpace);
+            
+            switch (direction)
+            {
+                case Direction.Left:
+                    MainCharacterCoordinates = (MainCharacterCoordinates.x, MainCharacterCoordinates.y - 1);
+                    break;
+                case Direction.Right:
+                    MainCharacterCoordinates = (MainCharacterCoordinates.x, MainCharacterCoordinates.y + 1);
+                    break;
+                case Direction.Up:
+                    MainCharacterCoordinates = (MainCharacterCoordinates.x - 1, MainCharacterCoordinates.y);
+                    break;
+                case Direction.Down:
+                    MainCharacterCoordinates = (MainCharacterCoordinates.x + 1, MainCharacterCoordinates.y);
+                    break;
+            }
+            
+            Console.SetCursorPosition(MainCharacterCoordinates.y, MainCharacterCoordinates.x);
+            Console.Write('@');
+        }
+    }
+
+    public void GoLeft(object? sender, EventArgs args)
+    {
+        MoveMainCharacter(Direction.Left);
+    }
+    public void GoRight(object? sender, EventArgs args)
+    {
+        MoveMainCharacter(Direction.Right);
+    }
+    public void GoUp(object? sender, EventArgs args)
+    {
+        MoveMainCharacter(Direction.Up);
+    }
```

Comment:
```suggestion
    }
    
```
Спейсинг между методами нужно соблюдать.